Oraimunka24015004/src/AlgorithmicRecap.java

```diff
@@ -0,0 +1,172 @@
+import java.math.BigInteger;
+import java.util.*;
+
+public class AlgorithmicRecap {
+    public String taskOneReverts(String text) {
+        String result = "";
+        for (int i = text.length(); i >= 1; i--) {
+            char charcter = text.charAt(i - 1);
+            result = result + charcter;
+        }
+        return result;
+    }
+
+    boolean taskTwoIsPalindrome(String text) {
+        return text.equals(taskOneReverts(text));
+    }
+
+    boolean taskThreeIsAnagram(String text, String text2) {
+        Map<Character, Integer> map = getCharacters(text);
+        Map<Character, Integer> map2 = getCharacters(text2);
+        return map.equals(map2);
+    }
+
+    private HashMap<Character, Integer> getCharacters(String text) {
+        HashMap<Character, Integer> map = new HashMap<>();
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (map.containsKey(character)) {
+                map.put(character, map.get(character) + 1);
+            } else {
+                map.put(character, 1);
+            }
+        }
+        return map;
+    }
+
+    BigInteger taskFourCountNumbers(String text) {
+        BigInteger result = BigInteger.valueOf(0);
+        for (int i = 0; i < text.length(); i++) {
+            Character character = text.charAt(i);
+            if (Character.isDigit(character)) {
+                result = result.add(BigInteger.valueOf(Character.getNumericValue(character)));
+            }
+        }
+        return result;
+    }
+
+    int taskFiveGetFactorial(int number) {
+        int result = 1;
+        for (int i = 1; i <= number; i++) {
+            result *= i;
+        }
+        return result;
+    }
+
+    char taskSixGetUniqueChar(String text) {
+        Map<Integer, Character> map = new HashMap<>();
```

Comment:
why are we assigning characters to integers? what am i missing? please explain

Reply:
I don't why i did that :D but yes, thats not a the best solution